packages/server/modules/workspaces/repositories/projectRegions.ts

```diff
@@ -577,3 +587,66 @@ export const copyProjectWebhooksFactory =
 
     return copiedWebhookCountByProjectId
   }
+
+/**
+ * Copies rows from the following tables:
+ * - file_uploads
+ * - blob_storage
+ * Also copies blobs in storage from one region to the other.
+ */
+export const copyProjectBlobs =
+  (deps: {
+    sourceDb: Knex
+    sourceObjectStorage: ObjectStorage
+    targetDb: Knex
+    targetObjectStorage: ObjectStorage
+  }): CopyProjectBlobs =>
+  async ({ projectIds }) => {
+    const copiedBlobsCountByProjectId: Record<string, number> = {}
+
+    // Copy `blob_storage` table rows in batches
+    const selectBlobs = tables
+      .blobStorage(deps.sourceDb)
+      .select('*')
+      .whereIn(BlobStorage.col.streamId, projectIds)
+
+    for await (const blobs of executeBatchedSelect(selectBlobs)) {
+      // Write `blob_storage` rows to target db
+      await tables.blobStorage(deps.targetDb).insert(blobs).onConflict().ignore()
+
+      for (const blob of blobs) {
+        copiedBlobsCountByProjectId[blob.streamId] ??= 0
+        copiedBlobsCountByProjectId[blob.streamId]++
+
+        // Copy file blob from one regional storage to the other
+        const objectKey = getObjectKey(blob.streamId, blob.id)
+        const sourceBlob = await deps.sourceObjectStorage.client.send(
+          new GetObjectCommand({
+            Bucket: deps.sourceObjectStorage.bucket,
+            Key: objectKey
+          })
+        )
+        await deps.targetObjectStorage.client.send(
+          new PutObjectCommand({
+            Bucket: deps.targetObjectStorage.bucket,
+            Key: objectKey,
+            Body: sourceBlob.Body,
+            ContentLength: sourceBlob.ContentLength
+          })
+        )
+      }
+    }
```

Comment:
verify if we can use the pipe method

Reply:
sources appear to say that the aws sdk handles this for us, but TIL !